### PR TITLE
OpenMediaVault widget

### PR DESCRIPTION
**Pull request by userXinos**
## Proposed change

Widget for integration with OpenMediaVault (https://github.com/benphelps/homepage/discussions/1204)

изображение

```yaml
- OpenMediaVault:
    - Services:
        widget:
          type: openmediavault
          url: http://
          username: admin
          password: pass
          method: services.getStatus
    - S.M.A.R.T.:
        widget:
          type: openmediavault
          url: http://
          username: admin
          password: pass
          method: smart.getListBg
    - Downloader Plugin:
        widget:
          type: openmediavault
          url: http://
          username: admin
          password: pass
          method: downloader.getDownloadList
```

OMV RPC docs: https://docs.openmediavault.org/en/6.x/development/tools/omv_rpc.html
Real docs: [Source Code on GH](https://github.com/openmediavault/openmediavault/tree/master/deb/openmediavault/usr/share/openmediavault/engined/rpc)

## Type of change

- [x] New service widget
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (please explain)

## Checklist:

- [X] If adding a service widget or a change that requires it, I have added a corresponding PR to the documentation:[https://github.com/benphelps/homepage-docs/pull/140](https://github.com/benphelps/homepage-docs/pull/140) 
- [X] If adding a new widget I have reviewed the [guidelines](https://gethomepage.dev/en/more/development/#service-widget-guidelines).
- [X] If applicable, I have checked that all tests pass with e.g. `pnpm lint`.
- [X] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.

### CODE REVIEW

**Comment by userXinos** [comment]
I would like to discuss which widget methods should be implemented next. I don’t see any reason to implement CPU Utilization, Memory from the dashboard, since the glances widget provides more functions and an excellent implementation, plus both widgets return one method from the back end, i.e. if you could dine them into a group so as not to spam with a FULL_SYS_INFO request ...

**Comment by userXinos** [comment]
When Homepage TypeScript

**Comment by benphelps** [comment]
> When Homepage TypeScript

When someone does all the work and makes a PR for it, and has enough interest. We'd have to come to some agreement on what required engagement from the community to constitute a "majority" looks like.


Then given a "majority" of the community is interested in converting to TS; give the members a few weeks to review it, test it, etc., which would require a notice to freeze development until the review is complete and merged or otherwise.

It would be a tremendous amount of effort, esp. given the number of widgets currently under maintenance.

**Comment by userXinos** [comment]
I just asked in the plan if there were any plans for this or something like that without pretensions.

**Review by shamoon — CHANGES_REQUESTED** [review]
Thanks for the PR. A couple things:

- The widget should use normal "blocks", not implement its own style. We've come upon this issue before, but we prioritize consistency.
- Please revert the linting changes, it sorta muddies this PR

Also, since OMV setup isn't so trivial, if you could please provide some samples of API endpoints for testing, thanks.

**Comment by userXinos** [comment]
изображение

1) I just copied the style from the OMV dashboard, which widget should I pay attention to to implement this functionality? Or to do the normal text list ?

**Comment by userXinos** [comment]
[http.zip](https://github.com/benphelps/homepage/files/12327586/http.zip)

**Comment by shamoon** [comment]
Thanks. Yea the text "block" widgets.

**Comment by userXinos** [comment]
изображение

Current state. ~~Not so much fun~~

**Comment by shamoon** [comment]
Sorry, to clarify, they should be some version of e.g:, just like the downloader one
```
    <Container service={service}>
       <Block label=..." value={...} />
       <Block label="..." value={...} />
     </Container>
```

This is how fields filtering works and its the standard for all service widgets.

I'll leave it up to you how to handle that, if you list the number of drives with SMART OK, services up / total, or whatever. Personally I would merge the widgets into one and support different fields (with a default limited to max 4 as in the guidelines). An example like this is the gamedig widget: https://github.com/benphelps/homepage/pull/1717/files

**Comment by userXinos** [comment]
alright

**Comment by userXinos** [comment]
изображение

**Review by shamoon — APPROVED** [review]
Thanks for your flexibility, this is great overall. I made some relatively minor changes (some needed, like placing error handling before the items manipulation, unique translation strings, etc). Please let me know if you have any concerns

Screenshot 2023-08-13 at 8 19 22 AM
Screenshot 2023-08-13 at 8 18 46 AM
Screenshot 2023-08-13 at 8 09 47 AM

File: src/widgets/openmediavault/components/total_numbers.jsx
Comment (userXinos) [inline src/widgets/openmediavault/components/total_numbers.jsx:1]: why not

Comment (shamoon) [reply src/widgets/openmediavault/components/total_numbers.jsx:1]: To put the error output above the items.reduce calls. I also just find it a bit less convoluted. I realize of course there is some code repetition but minimal.

**Comment by userXinos** [comment]
I was hoping that the members of the discussion would like to see some of the methods released. I will implement a few later. Can I use the graphics widget style from Glances Widget for this ?
изображение

**Comment by shamoon** [comment]
Yes you can use the charts or the standard ones. Im happy to merge this in the meantime if you’re good with it

**Comment by Mbarmem** [comment]
> getInformation

Hi, would you mind sharing the code for pulling CPU & Memory usage.

I am using OMV on a different system and would love to implement it.

**Comment by Mbarmem** [comment]
> I would like to discuss which widget methods should be implemented next. I don’t see any reason to implement CPU Utilization, Memory from the dashboard, since the glances widget provides more functions and an excellent implementation, plus both widgets return one method from the backend, i.e. if you could dine them into a group so as not to spam with a FULL_SYS_INFO request ... .
> 
> Network Developer Tools: `[POST] {"service":"System","method":"getInformation"...}` изображение

Hi, would you mind sharing the code for pulling CPU & Memory usage.

I am using OMV on a different system and would love to implement it.

**Comment by userXinos** [comment]
@Mbarmem 

I don't have such code, can you see something similar in the glances widget?

**Pull request merged**